packages/block/src/header.ts

```diff
@@ -99,6 +99,15 @@ export class BlockHeader {
       throw new Error('Invalid serialized header input. Must be array')
     }
 
+    // If an RLP serialized block header is provided and no `hardforkByBlockNumber` opt is provided, default true to
+    // avoid scenarios where no `opts.common` or `opts.hardforkByBlockNumber` is provided and a serialized blockheader
+    // is provided that predates London result in a default base fee being added to the block
+    // (resulting in an erroneous block hash since the default `common` hardfork is London and the blockheader constructor
+    // adds a default basefee if EIP-1559 is active and no basefee is provided in the `headerData`)
+    if (opts.hardforkByTTD === undefined) {
```

Comment:
`hardforkByTTD` doesn't exist, this should be `hardforkByTD`.

Reply:
:facepalm: Ack.  Will add it to the fixes list.

Reply:
Wait, I have another comment... 😋

Reply:
I will soon write a more general comment on this solution itself, but *if* we would take the solution here, wouldn't we need to also check that `Common` is *not* provided? otherwise we'll overwrite in many cases (which is likely not intended by the user)? 🤔